### PR TITLE
feat(e2e): adding playwright and migrate landing page e2e tests

### PR DESCRIPTION
**Pull request by hassaanp**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -33813,15 +33824,15 @@ packages:
...
+      debug: 3.2.7(supports-color@5.5.0)
     transitivePeerDependencies:
       - supports-color
     dev: true
 
   /webpack-virtual-modules@0.3.2:
     resolution: {integrity: sha512-RXQXioY6MhzM4CNQwmBwKXYgBs6ulaiQ8bkNQEl2J6Z+V+s7lgl/wGvaI/I0dLnYKB8cKsxQc17QOAVIphPLDw==}
     dependencies:
-      debug: 3.2.7(supports-color@8.1.1)
+      debug: 3.2.7(supports-color@5.5.0)
```
Comment (hassaanp) [inline pnpm-lock.yaml]: this should be reverted - leaving this here to remember it

File: e2e/landing.spec.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    if (!isMobile) {
+      const landingPageImage = page.getByTestId(
+        landingPageElements.landingPageImage
+      );
+
+      expect(await landingPageImage.isVisible()).toBeTruthy();
+    }
+  });
+
+  // test('Has `certifications` section', async ({ page }) => {
```
Comment (mirrzamustafa) [inline e2e/landing.spec.ts]: we can either remove or uncomment this test case

Comment (hassaanp) [reply e2e/landing.spec.ts]: This is a work in progress.

File: playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    //   use: { ...devices['Desktop Safari'] }
+    // },
+
+    /* Test against mobile viewports. */
+    {
+      name: 'Mobile Chrome',
+      use: { ...devices['Pixel 5'] }
+    }
+    // {
+    //   name: 'Mobile Safari',
```
Comment (mirrzamustafa) [inline playwright.config.ts]: can we uncomment these to make the tests more extensive i.e: multiple viewports and browsers?

File: playwright.config.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    //   use: { ...devices['Desktop Edge'], channel: 'msedge' }
+    // },
+    // {
+    //   name: 'Google Chrome',
+    //   use: { ...devices['Desktop Chrome'], channel: 'chrome' }
+    // }
+  ]
+
+  /* Run your local dev server before starting the tests */
+  // webServer: {
```
Comment (mirrzamustafa) [inline playwright.config.ts]: I think we can safely remove this local server comment block.

**Review by mirrzamustafa — APPROVED** [review]

File: playwright.config.ts (hunk 3 above)
Comment (hassaanp) [reply playwright.config.ts]: webkit is crashing on my system
can someone test the mobile and safari browsers @Ahmadkashif

File: playwright.config.ts (hunk 4 above)
Comment (hassaanp) [reply playwright.config.ts]: I'd like to keep it for now - to allow people to tinker.

File: e2e/landing.spec.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    await page.goto('/');
+  });
+
+  test('Should render', async ({ page }) => {
+    await expect(page).toHaveTitle(
+      'Learn to Code — For Free — Coding Courses for Busy People'
+    );
+
+    const callToAction = page.getByTestId(landingPageElements.callToAction);
+    // const callToActionHeader = page.getByTestId(landingPageElements.callToAction).nth(1);
```
Comment (hassaanp) [inline e2e/landing.spec.ts]:
```suggestion
```

**Comment by Ahmadkashif** [comment]
Closing this as the file discussed here has been merged on FCC main

**Pull request closed without merging**